sklearn/feature_selection/_univariate_selection.py

```diff
@@ -294,15 +294,34 @@ def f_regression(X, y, center=True):
     else:
         X_norms = row_norms(X.T)
 
+    # the Pearson correlation coefficient is undefined
+    # for random variables (regressors and target) with
+    # a standard deviation of zero
+    if np.any(X_norms == 0) or np.std(y) == 0:
+        raise ValueError("The correlation is undefined for "
+                         "regressors and target with "
```

Comment:
for some reason I cannot suggest edits so:

regressors -> features
and -> or  (or "and / or")

Reply:
I prefer "features" instead of "regressor" too.

Fixed!